adal/src/main/java/com/microsoft/aad/adal/Logger.java

```diff
@@ -154,163 +194,59 @@ private static String addMoreInfo(String message) {
     }
 
     /**
-     *
-     * @param tag TAG for the log message
-     * @param message Body of the message
+     * Format the log message. Depends on the developer setting, the log message could be sent to logcat
+     * or the external logger set by the calling app.
      */
-    public void debug(String tag, String message) {
-        if (mLogLevel.compareTo(LogLevel.Debug) < 0 || StringExtensions.isNullOrBlank(message)) {
-            return;
-        }
-
-        if (mAndroidLogEnabled) {
-            Log.d(tag, message);
-        }
-
-        logCommon(tag, message, "", LogLevel.Info, null);
-    }
-
-    /**
-     *
-     * @param tag tag for the log message
-     * @param message body of the log message
-     * @param additionalMessage additional parameters
-     * @param errorCode ADAL error code being logged
-     */
-    public void verbose(String tag, String message, String additionalMessage, ADALError errorCode) {
-        if (mLogLevel.compareTo(LogLevel.Verbose) < 0) {
+    private void log(String tag, String message, String additionalMessage, LogLevel logLevel,
+                           ADALError errorCode, Throwable throwable) {
+        if (logLevel.compareTo(mLogLevel) > 0) {
             return;
         }
 
-        if (mAndroidLogEnabled) {
-            Log.v(tag, getLogMessage(message, additionalMessage, errorCode));
-        }
+        // Developer turns off PII logging, if the log message contains any PII, we shouldn't send it.
 
-        logCommon(tag, message, additionalMessage, LogLevel.Verbose, errorCode);
-    }
+        final StringBuilder logMessage = new StringBuilder();
 
-    /**
-     *
-     * @param tag tag for the log message
-     * @param message body of the log message
-     * @param additionalMessage additional parameters
-     * @param errorCode ADAL error code being logged
-     */
-    public void inform(String tag, String message, String additionalMessage, ADALError errorCode) {
```

Comment:
was all this dead code?

Reply:
Yes, they are all dead now. I replace all these functions with loglevel parameter of function log().